barretenberg/cpp/src/barretenberg/vm2/simulation/bytecode_manager.hpp

```diff
@@ -98,8 +94,14 @@ class BytecodeManagerInterface {
     virtual ~BytecodeManagerInterface() = default;
 
     virtual Instruction read_instruction(uint32_t pc) = 0;
+
     // Returns the id of the current bytecode. Tries to fetch it if not already done.
+    // Throws BytecodeNotFoundError if contract not found (for use in temporality group 1)
     virtual BytecodeId get_bytecode_id() = 0;
+
+    // Returns the id of the current bytecode, or nullopt if contract not found.
```

Comment:
Note to self: "if contract not found."

Does this try to fetch the contract? Or "contract not found" means "nobody tried to fetch yet"?